src/customer/strategies/googlepay/googlepay-authorizenet-customer-strategy.spec.ts

```diff
@@ -208,23 +208,40 @@ describe('GooglePayCustomerStrategy', () => {
     });
 
     describe('#handleWalletButtonClick', () => {
-        it('handles wallet button event', async () => {
-            customerInitializeOptions = {
-                methodId: 'googlepayauthorizenet',
-                googlepayauthorizenet: {
-                    container: 'googlePayCheckoutButton',
-                },
-            };
+        const googlePaymentDataMock = getGooglePaymentDataMock();
 
-            jest.spyOn(paymentProcessor, 'displayWallet').mockReturnValue(Promise.resolve(getGooglePaymentDataMock()));
+        beforeEach(() => {
+            customerInitializeOptions = getAuthNetCustomerInitializeOptions();
+
+            jest.spyOn(paymentProcessor, 'displayWallet').mockResolvedValue(googlePaymentDataMock);
             jest.spyOn(paymentProcessor, 'handleSuccess').mockReturnValue(Promise.resolve());
             jest.spyOn(paymentProcessor, 'updateShippingAddress').mockReturnValue(Promise.resolve());
+        });
 
-            expect(paymentProcessor.initialize).not.toHaveBeenCalled();
+        it('handles wallet button event and updates shipping address', async () => {
             await strategy.initialize(customerInitializeOptions);
 
             walletButton.click();
-            expect(paymentProcessor.initialize).toHaveBeenCalledWith('googlepayauthorizenet');
+
+            await new Promise(resolve => process.nextTick(resolve));
+
+            expect(paymentProcessor.displayWallet).toHaveBeenCalled();
+            expect(paymentProcessor.handleSuccess).toHaveBeenCalledWith(googlePaymentDataMock);
+            expect(paymentProcessor.updateShippingAddress).toHaveBeenCalledWith(googlePaymentDataMock.shippingAddress);
+        });
+
+        it('handles wallet button event and does not update shipping address if cart has digital products only', async () => {
```

Comment:
Since we are already grouping in '#handleWalletButtonClick' 🍹

```suggestion
        it('displays the wallet and does not update the shipping address if cart has digital products only', async () => {
```